src/Generation/Generator/Renderer/Public/ReturnType/From/IRenderableReturnType.cs

```diff
@@ -0,0 +1,7 @@
+﻿namespace Generator.Renderer.Public;
+
+internal interface IRenderableReturnType
+{
+    RenderableReturnType Create(GirModel.ReturnType returnType);
+    bool Supports(GirModel.ReturnType returnType);
```

Comment:
Yeah, I wasn't really sure what was best here. On one hand, having the extra info from the `ReturnType` object could be useful in some cases. On the other, there is perhaps a slight cognitive overhead associated with all of the `.AsT0` and `.AsT1` littered throughout the various implementations of the `Create()` method. And there's also the fact that mistakes or inconsistencies between the implementation of `Supports()` and `Create()` would lead to problems which get revealed at runtime, rather than compile-time.

But probably not super important either way.

Reply:
Yeah there could be an inconsistency as both methods need to match each other.

If there is some bug reported during runtime I would suggest to write a unit test in this moment which makes this problem visible and afterwards fix the code so the problem should not be able to return.

Writing such tests beforehand is just to much work if the project is done in spare time (at least for me). Therefore it is okay for me to omit tests here as the classes are pretty simple in itself.